### PR TITLE
iaac components

### PR DESCRIPTION
**Pull request by luiztauffer**
WIP
IaaC components

### CODE REVIEW

File: python/dendro/iaac/aws_batch_job_definition/job_definition.py [hunk 1]
```diff
@@ -64,16 +63,22 @@ def create_job_definition(
...
         {
-            'containerPath': 'string',
+            'containerPath': '/mount/efs',
             'readOnly': False,
             'sourceVolume': f'{job_definition_name}-EfsVolume'
         },
     ]
 
+    # Container Command
+    container_command = ["python", "main.py"]
```
Comment (magland) [inline python/dendro/iaac/aws_batch_job_definition/job_definition.py]: @luiztauffer Should this be "/app/main.py" instead of just "main.py"?

Comment (luiztauffer) [reply python/dendro/iaac/aws_batch_job_definition/job_definition.py]: @magland this is just a placeholder for now, but sure, we should use the default command for dendro apps
fyi, this is something we can override in two occasions:
- when creating the Job Definition, using the `container_command_override` arg of this function
- when submitting jobs

File: python/dendro/iaac/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+1. Before installing any Apps: Provision the base AWS Batch infrastructure using CDK. This includes IAM roles, VPC, Security Group, EFS filesystems, Batch Compute Environments and Batch Job Queues.
+2. At every App install: Provision the App-specific infrastructure using Boto3. This includes one EFS Volume one Batch Job Definition.
+
+## Prerequisites
+
+- Python 3.10
+- [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html)
+- [AWS CDK Toolkit](https://docs.aws.amazon.com/cdk/latest/guide/cli.html)
+- [AWS CDK Python](https://docs.aws.amazon.com/cdk/latest/guide/work-with-cdk-python.html)
```
Comment (magland) [inline python/dendro/iaac/README.md]: Is this something that requires user action? I followed the link and I couldn't see where it actually instructed to install aws_cdk.

File: python/dendro/iaac/README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+- Set up your AWS credentials, either by [configuring the AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html) or by [setting environment variables](https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-envvars.html).
```
Comment (magland) [inline python/dendro/iaac/README.md]: I followed this link for the CLI but it wasn't clear at all what I should do. It's strange that they would call that guide "quickstart.html".

File: python/dendro/iaac/README.md (hunk 2 above)
Comment (magland) [reply python/dendro/iaac/README.md]: Looks like you need to run `pip install -r requirements.txt` in the aws_batch dir.

File: python/dendro/iaac/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+
+## Provision base Batch infrastructure with CDK
+
+AWS CDK is a very convenient tool, it helps automate the provisioning of AWS infrastructure and organizes all created resources in a CloudFormation stack, which can be easily updated or deleted.
+
+Follow these steps if you're running Dendro CDK stack for the first time:
+- go to `cloned_dendro_repo_path/python/dendro/iaac/aws_batch` directory.
+- run `cdk bootstrap` to set up the CDK deployment infrastructure (CKDToolkit) in your AWS account. This only needs to be done once.
```
Comment (magland) [inline python/dendro/iaac/README.md]: Looks like you need to do `mkdir ~/.dendro`

File: python/dendro/iaac/README.md (hunk 3 above)
Comment (magland) [reply python/dendro/iaac/README.md]: I ended up generating access keys and then running `aws configure` but it was difficult to figure out to do that.

File: python/dendro/iaac/README.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+- run `cdk synth` to synthesize the CloudFormation template.
```
Comment (magland) [inline python/dendro/iaac/README.md]: I got the following error
```bash
AssertDescription: CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.
```

Comment (magland) [reply python/dendro/iaac/README.md]: I am not sure how to upgrade cdk cli. I am proceeding anyway, assuming the error wasn't failure.

File: python/dendro/iaac/README.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+- run `cdk deploy` to deploy the stack to your default AWS account/region.
```
Comment (magland) [inline python/dendro/iaac/README.md]: I get the following error on running `cdk deploy`
```bash
DendroBatchStack failed: Error [ValidationError]: Template error: Unable to get mapping for DendroBatchStackcomputeenvAmiMap7E2800BD::us-east-1::ami
...

```

Comment (magland) [reply python/dendro/iaac/README.md]: This has been resolved by adding us-east-1 to machine_image in aws_batch_stack.py

**Comment by magland** [comment]
@luiztauffer I made some minor commits to get it to pass the tests.

**Pull request merged**